dist/gcc-compatible/EverCrypt_Hash.h

```diff
@@ -116,9 +118,9 @@ called EverCrypt_AutoConfig2_init() before.
 void
 EverCrypt_Hash_Incremental_hash(
   Spec_Hash_Definitions_hash_alg a,
```

Comment:
Shouldn't the `output` come first?

Reply:
I treated the "algorithm" input as a special input, like "state" in the streaming API, thus letting it at the beginning

Reply:
The state is special because it's really an i/o parameter. This is a regular input imo. Has there been a discussion for leaving it in front here? Not sure what others think (@msprotz?).

Reply:
The state is special because it's really an i/o parameter. This is a regular input imo. Has there been a discussion for leaving it in front here? Not sure what others think (@msprotz?).

Reply:
honestly I don't really have much of an opinion here. I think agile APIs take the agility parameter first because of the original type in F*, of the form:

```
let hash_t (a: alg) = output:hash a -> input:buffer uint8 -> len: uint32 { len == length input }
val hash #a: hash_t a
let hash #a output input len = ...
```

so you might have a hard time changing that without making signature very difficult to deal with in the source (we can, but is it worth it?)